src/components/src/test/java/org/apache/jmeter/assertions/jmespath/TestJMESPathAssertion.java

```diff
@@ -0,0 +1,207 @@
+                    { InvertType.USE_NO_INVERT,
+                            "{\n" + "  \"reservations\": [\n" + "    {\n" + "      \"instances\": [\n"
+                                    + "        {\"state\": \"running\"},\n" + "        {\"state\": \"stopped\"}\n"
+                                    + "      ]\n" + "    },\n" + "    {\n" + "      \"instances\": [\n"
+                                    + "        {\"state\": \"terminated\"},\n" + "        {\"state\": \"running\"}\n"
+                                    + "      ]\n" + "    }\n" + "  ]\n" + "}",
+                            "reservations[*].instances[*].state", ValidationType.USE_VALIDATION,
+                            ComparisonType.USE_NO_REXEG, ResultNullity.EXPECT_NOT_NULL,
+                            "[[\"running\",\"stopped\"],[\"terminated\",\"running\"]]", ResultType.SUCCESS, "" } });
+        }
+
+
+        private InvertType isInverted;
+        private String responseData;
+        private String jmesPath;
+        private ValidationType isValidation;
+        private ComparisonType isRegex;
+        private ResultNullity isExpectedNull;
+        private String expectedValue;
+        private ResultType resultType;
+        private String failureMessage;
+
+        public TestAssertion(InvertType isInverted, String responseData, String jmesPath, ValidationType isValidation, ComparisonType isRegex,
+                ResultNullity isExpectedNull, String expectedValue, ResultType resultType, String failureMessage) {
+            super();
+            this.isInverted = isInverted;
+            this.responseData = responseData;
+            this.jmesPath = jmesPath;
+            this.isValidation = isValidation;
+            this.isRegex = isRegex;
+            this.isExpectedNull = isExpectedNull;
+            this.expectedValue = expectedValue;
+            this.resultType = resultType;
+            this.failureMessage = failureMessage;
+        }
+
+        @Test
+        public void test() {
+            SampleResult samplerResult = new SampleResult();
+            samplerResult.setResponseData(responseData, null);
+            JMESPathAssertion instance = new JMESPathAssertion();
+            instance.setJmesPath(jmesPath);
+            instance.setJsonValidationBool(isValidation == ValidationType.USE_VALIDATION);
+            instance.setInvert(isInverted == InvertType.USE_INVERT);
+            instance.setIsRegex(isRegex == ComparisonType.USE_REGEX);
+            instance.setExpectNull(isExpectedNull == ResultNullity.EXPECT_NULL);
+            instance.setExpectedValue(expectedValue);
+            AssertionResult expResult = new AssertionResult("");
+            AssertionResult result = instance.getResult(samplerResult);
+            assertEquals(expResult.getName(), result.getName());
+            switch (resultType) {
+                case SUCCESS:
+                    assertFalse(result.isError());
+                    assertFalse(result.isFailure());
```

Comment:
Asserttrue/false produce hard-to-understand error messages.

It would be better if you could convert `result.isError()+result.isFailure()` into `ResultType` enum, and assert on the expected value of that enum.